src/boxed.rs

```diff
@@ -366,10 +366,10 @@ impl<T: 'static, MM: BoxedMemoryManager<T>> Uninitialized for Boxed<T, MM> {
     unsafe fn uninitialized() -> Self {
         Boxed {
             inner: {
-                let mut inner = Box::<T>::new(mem::zeroed());
-                MM::init(&mut *inner);
+                let mut inner = Box::new(mem::MaybeUninit::zeroed());
+                MM::init(inner.as_mut_ptr());
 
-                AnyBox::Native(inner)
+                AnyBox::Native(Box::new(inner.assume_init()))
```

Comment:
You're creating two boxes here now instead of one. Maybe just

```rust
let mut inner = mem::MaybeUninit::zeroed();
MM::init(inner.as_mut_ptr());
AnyBox::Native(Box::new(inner.assume_init()))
```
?